clients/include/hipsolver_fortran.f90

```diff
@@ -153,6 +153,167 @@ function hipsolverZgeqrfFortran(handle, m, n, A, lda, tau, work, lwork, info) &
         res = hipsolverZgeqrf(handle, m, n, A, lda, tau, work, lwork, info)
     end function hipsolverZgeqrfFortran
 
+    ! ******************** GESVD ********************
+    function hipsolverSgesvd_bufferSizeFortran(handle, m, n, lwork) &
+            result(res) &
+            bind(c, name = 'hipsolverSgesvd_bufferSizeFortran')
+        use iso_c_binding
+        use hipsolver_enums
+        implicit none
+        type(c_ptr), value :: handle
+        integer(c_int), value :: m
+        integer(c_int), value :: n
+        type(c_ptr), value :: lwork
+        integer(c_int) :: res
+        res = hipsolverSgesvd_bufferSize(handle, m, n, lwork)
+    end function hipsolverSgesvd_bufferSizeFortran
```

Comment:
This is more a generic question about hipsolver than one specific to SVD, but does anything use the Fortran API? If not, we might want to add a Fortran sample.

Reply:
By "anything", do you mean client code? The client code tests it in an extremely roundabout way. The tests will call functions in hipsolver.hpp with a FORTRAN bool, which will call functions in hipsolver_fortran.hpp with a Fortran suffix, which extern to Fortran code in hipsolver_fortran.f90, which itself will call the library Fortran code in hipsolver_module.f90.

I guess the intent is specifically to make sure we can call our Fortran code from other Fortran code - everything else is just the C++ infrastructure to get us to that point. A sample would have to be written in Fortran as well, which I wouldn't know how to do.